runtime/lua/lsp/structures.lua

```diff
@@ -0,0 +1,117 @@
+local lsp_util = require('runtime.lua.lsp.util')
```

Comment:
drive-by comment: `runtime.lua.lsp.util` looks too nested, possibly a sign of over-architecture. Hyper-categorization is bad enough, but sub-categories and sub-sub-categories are rarely needed. 

I think we already have too many subdirectories in Nvim source. We should try to keep things relatively flat.

Reply:
In fact the LSP support should be 1 big file, with truly _common_ utilities in a separate file (call it `foo.lua`, doesn't matter: again, 1 big file, we can nitpick over naming/placement later). "Common" means "can be used by any other Lua code, not just LSP".

Reply:
The most important thing to avoid is shared state. The line count of a file doesn't mean anything as long as there is minimal or no shared global or pseudo-global state.

Reply:
I think it looks to nested only because it's wrong, it should be `require('lsp.util')`. That wouldn't seem too bad?

Reply:
Yeah, it should just be `lsp.util`, but I couldn't get the tests working at first without the `runtime.lua`, so I'll fix that soon.

Reply:
I am looking to try and consolidate as much as possible :)

Reply:
@justinmk Should be fixed now. It's just `lsp.util` or `lsp.callbacks`, just like any other lua plugin.